src/parsing.jl

```diff
@@ -244,6 +244,51 @@ end
     return block
 end
 
+###
+### Buses
+###
+
+# In v30 files, we've seen trailing end-of-line comments. These can be present
+# in any section, but so far we have only had a use-case for the comments in the
+# buses section.
+# See https://github.com/nickrobinson251/PowerFlowData.jl/issues/27
+#
+# The currently handles data which looks like:
+#  111,'STBC      ',161.00,1,    0.00,    0.00,227,   1,1.09814,  -8.327,  1 /* [STBC   1   ] */
+# And does _not_ handle data with a comma before the comment like:
+#  111,'STBC      ',161.00,1,    0.00,    0.00,227,   1,1.09814,  -8.327,  1, /* [STBC   1   ] */
+@generated function parse_row!(rec::R, bytes, pos, len, options) where {R <: Buses30}
```

Comment:
unfortunately thiis is the necessary to avoid dynamic dispatch (https://github.com/nickrobinson251/PowerFlowData.jl/issues/22)

whenever we call:
```julia
fo col in 1:ncols
  T = eltype(fieldtype(R, col))
  parse_value!(record, col, T, bytes, pos, len, options)
```
the type of `T` is not known til runtime, meaaning the `parse_value!` call has to look-up decide at runtime which method to call, which adds a _big_ performance cost, since we have one `parse_value!` call per value/element in the file

so we use a generated function to unroll the loop and generate code like
```julia
parse_value!(record, 1, Int64, bytes, pos, len, options)
parse_value!(record, 2, Float64, bytes, pos, len, options)
...
parse_value!(record, 12, Bool, bytes, pos, len, options)
```

I think the other way we might have been able to avoid the dynamic dispatch is by manually doing [what "union splitting" does](https://julialang.org/blog/2018/08/union-splitting/) e.g. like is [done in ODBC.jl](https://github.com/JuliaDatabases/ODBC.jl/blob/f38f771557a5763f777ed0f441c2cc5a72f41c70/src/utils.jl#L28-L80)

CSV.jl uses a mix of these techniques e.g. [splitting the big union of expected types](https://github.com/JuliaData/CSV.jl/blob/60736525b243c8d02fd54897ca1714d7e10d4043/src/file.jl#L588-L645) and [generating code for custom types](https://github.com/JuliaData/CSV.jl/blob/60736525b243c8d02fd54897ca1714d7e10d4043/src/file.jl#L847-L869)